Sources/ViewStateMachine.swift

```diff
@@ -226,6 +207,10 @@ public class ViewStateMachine {
 		let animationCompletion: (Bool) -> Void = { [weak self] _ in
 			for (_, view) in store {
 				view.removeFromSuperview()
+
```

Comment:
I think this isn't needed, as the callback closure you've configured above has a weak reference to instance. So once the placeholder view is deallocated, the closure will be deleted as well.

But… better safe than sorry 😉